src/engine/Buffer.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
```

Comment:
Why does this class need to exist? Couldn't everything here be done by the ResourceManager?